core/payments/reservation/reservation.go

```diff
@@ -0,0 +1,96 @@
+package reservation
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/core"
+)
+
+// Represents a reservation for a single account.
+//
+// TODO(litt3): I opted to duplicate the preexisting `ReservedPayment` struct, rather than using the old one. There
+// are nontrivial changes I wanted to make, and making those changes in a way that's compatible with the preexisting
+// usages was going to be messy. Instead, `ReservedPayment` can just be removed, when we remove the deprecated payment
+// system.
+type Reservation struct {
+	// The number of symbols / second that the holder of this reservation is entitled to disperse
+	symbolsPerSecond uint64
+
+	// The time at which the reservation becomes active
+	startTime time.Time
+
+	// The time at which the reservation expires
+	endTime time.Time
+
+	// The quorums that the holder of this reservation is entitled to disperse to
+	permittedQuorumIDs map[core.QuorumID]bool
```

Comment:
```suggestion
	permittedQuorumIDs map[core.QuorumID]struct{}
```

I prefer `struct{}` instead of `bool` for creating the abstraction of a set. I'm under the impression that it's slightly more efficient (not that the difference makes a meaningful difference here). Additionally, it very clearly marks it as a set, as there is no way to convey any information in the map values.

Reply:
23c1d7a31d584af6f8824d6614b70204cb1546ec